### PR TITLE
Refactor anonymous async functions into arrow-functions

### PR DESCRIPTION
**Pull request by ctgardner**
Fixes https://github.com/mozilla/multi-account-containers/issues/972

I've refactored all anonymous async functions declared using the `function` keyword into arrow-functions.

I understand the crux of this issue isn't to simply replace functions declared using the `function` keyword with arrow-functions, but to inline named async functions where sensible. I've reviewed all named async functions and I believe none would benefit from being inline, with the exception of a few handlers registered using `addListener`. Given the past confusion surrounding this issue, I believe this issue should be closed.

### CODE REVIEW

File: test/issues/940.test.js [hunk 1]
```diff
@@ -160,7 +160,7 @@ describe("#940", () => {
       await promise2;
 
       background.browser.tabs.create.should.have.been.calledTwice;
-    }).timeout(2002);
```
Comment (jonathanKingston) [inline test/issues/940.test.js]: Why this change out of interest?

Comment (ctgardner) [reply test/issues/940.test.js]: This test was passing before the refactor and began failing after making the changes. As the test seems unrelated to the changes, I increased the timeout. (Two millis is the minimum increase necessary to make the test pass on my machine.)

Comment (ctgardner) [reply test/issues/940.test.js]: The test is passing on CI after reverting the timeout change 🤷‍♂

Comment (jonathanKingston) [reply test/issues/940.test.js]: Heh thanks!

**Comment by jonathanKingston** [comment]
Merged thanks for this!
